src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -7,297 +7,405 @@
+                                                                                     OpenLoadFlowParameters lfParametersExt) {
         Objects.requireNonNull(network);
         Objects.requireNonNull(factors);
         Objects.requireNonNull(lfParametersExt);
 
-        // create LF network (we only manage main connected component)
+        // create the network (we only manage main connected component)
         List<LfNetwork> lfNetworks = LfNetwork.load(network, lfParametersExt.getSlackBusSelector());
         LfNetwork lfNetwork = lfNetworks.get(0);
-
-        // run DC load
-        Map<String, Double> functionReferenceByBranch = new HashMap<>();
-        DcLoadFlowParameters dcLoadFlowParameters = new DcLoadFlowParameters(lfParametersExt.getSlackBusSelector(), matrixFactory,
-                true, lfParametersExt.isDcUseTransformerRatio(), lfParameters.isDistributedSlack(),  lfParameters.getBalanceType());
-        DcLoadFlowResult dcLoadFlowResult = new DcLoadFlowEngine(lfNetworks, dcLoadFlowParameters).run();
-        for (LfBranch branch : dcLoadFlowResult.getNetwork().getBranches()) {
-            functionReferenceByBranch.put(branch.getId(), branch.getP1() * PerUnit.SB);
-        }
+        checkContingencies(lfNetwork, contingencies);
+        checkSensitivities(network, lfNetwork, factors);
+        LazyConnectivity connectivity = new LazyConnectivity(lfNetwork);
 
         // create DC equation system for sensitivity analysis
         EquationSystem equationSystem = DcEquationSystem.create(lfNetwork, new VariableSet(),
                 new DcEquationSystemCreationParameters(false, true, true, lfParametersExt.isDcUseTransformerRatio()));
 
-        // index factors by variable configuration to compute minimal number of DC state
-        Map<SensitivityVariableConfiguration, SensitivityFactorGroup> factorsByVarConfig = indexFactorsByVariableConfig(network, factors, lfNetwork, lfParameters);
+        // we wrap the factor into a class that allows us to have access to their branch and EquationTerm instantly
+        List<SensitivityFactorWrapped> factorsWrapped = factors.stream().map(factor -> new SensitivityFactorWrapped(factor, lfNetwork, equationSystem)).collect(Collectors.toList());
 
-        if (factorsByVarConfig.isEmpty()) {
-            return Collections.emptyList();
+        // run DC load
+        setFunctionReferenceOnFactors(lfNetworks, factorsWrapped, lfParameters, lfParametersExt);
+
+        // index factors by variable configuration to compute minimal number of states
+        List<SensitivityFactorGroup> factorsGroups = createFactorGroups(network, factorsWrapped);
+
+        if (factorsGroups.isEmpty()) {
+            return Pair.of(Collections.emptyList(), Collections.emptyMap());
         }
 
-        // initialize right hand side
-        DenseMatrix rhs = initRhs(lfNetwork, equationSystem, factorsByVarConfig);
+        // Compute the partipation for every factor
```

Comment:
Improve comment: "// Compute the participation to slack distribution for every factor". I can make some improvements in a dedicated commit if you agree, it will be easier.

Reply:
Yes sure